crypto/cmp/cmp_vfy.c

```diff
@@ -394,21 +398,25 @@ static int check_kid(X509 *cert, const ASN1_OCTET_STRING *skid, int fn)
  *
  * Check if the given cert is acceptable as sender cert of the given message.
  * The subject DN must match, the subject key ID as well if present in the msg,
- * and the cert must not be expired (for checking this, the ts must be given).
+ * and the cert must be current (for checking this, the ts should be given).
+ * Note that cert revocation etc. is checked by OSSL_CMP_validate_cert_path().
  * returns 0 on error or not acceptable, else 1
  */
 static int cert_acceptable(X509 *cert, const OSSL_CMP_MSG *msg,
                            const X509_STORE *ts) {
     X509_NAME *sender_name = NULL;
     X509_VERIFY_PARAM *vpm = NULL;
+    int time_cmp;
 
     vpm = ts != NULL ? X509_STORE_get0_param((X509_STORE *)ts) : NULL;
-    if (cert == NULL || msg == NULL || ts == NULL || vpm == NULL)
+    if (cert == NULL || msg == NULL || (ts != NULL && vpm == NULL))
```

Comment:
why ts must be checked here? Checking vpm should be sufficiant?

Reply:
It would not be sufficient because `vpm == NULL` can be accepted if `ts == NULL`.